cbpfc_test.go

```diff
@@ -2,28 +2,38 @@ package cbpfc
 
 import (
 	"reflect"
+	"strings"
 	"testing"
 
 	"golang.org/x/net/bpf"
 )
 
+// requireError ensures an error is not nil, and it contains contains.
+func requireError(tb testing.TB, err error, contains string) {
+	tb.Helper()
+
+	if err == nil {
+		tb.Fatalf("expected error %s", contains)
+	}
+
+	if !strings.Contains(err.Error(), contains) {
```

Comment:
I'm not a fan. Either your error is so important that it's worth to export it to users, or you introduce an unexported error variable. Both can be checked by errors.Is. Checking error strings just makes tests explode when you decide to change the output, which is tedious for very little gain.

Reply:
The goal isn't to test / check the contents of the error so much as to check the test triggers the expected error. I find it very easy to create a test that I think will trigger a certain error, but in reality triggers an earlier one.
Adding unexported error variables everywhere just for tests feels a bit unnecessary to me, but I'm happy to do it if you feel strongly about it.

Reply:
It's your call, but in my opinion doing stringly matches on errors is not maintainable.

Reply:
Let's see how it goes. If it's too much work, I'll change it back or use some unexported errors :)